data-prepper-plugins/obfuscate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/obfuscation/ObfuscationProcessorConfig.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.obfuscation;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.Max;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
+
+public class ObfuscationProcessorConfig {
+
+    @JsonProperty("key")
+    @NotEmpty
+    @NotNull
+    private String key;
+
+    @JsonProperty("obfuscation_character")
+    private char character = '*';
+
+    @JsonProperty("obfuscation_length")
+    @Min(1)
+    @Max(10)
+    private int obfuscationLength = 3;
+
+
+    @JsonProperty("unobfuscated_length")
+    @Min(0)
+    private int unObfuscatedLength = 0;
+
+    @JsonProperty("pattern")
+    private String pattern = "";
+
+    @JsonProperty("new_key")
+    private String newKey = "";
```

Comment:
We usually keep this uninitialized and check for null.

Reply:
Sure, I will make the change.